src/server/rdb_load.cc

```diff
@@ -1902,6 +1915,32 @@ error_code RdbLoaderBase::HandleCompressedBlobFinish() {
   return kOk;
 }
 
+error_code RdbLoaderBase::HandleJournalBlob(Service* service, DbIndex dbid) {
+  // Read the number of entries in the journal blob.
+  size_t num_entries;
+  bool _encoded;
+  SET_OR_RETURN(LoadLen(&_encoded), num_entries);
+
+  // Read the journal blob.
+  string journal_blob;
+  SET_OR_RETURN(FetchGenericString(), journal_blob);
+
+  io::BytesSource bs{io::Buffer(journal_blob)};
+  JournalReader rd{&bs, dbid};
```

Comment:
In this flow JournalReader is created for each journal entry.
JournalReader has a private member buf_ which grows as you read the command, I think you have many allocations in this flow

Reply:
Indeed... I haven't thought about this at all - just wanted to get it working

But instead of having of adding a member `JournalReader`, I would rather make it re-use the `mem_buffer_` from rdb loader

Reply:
This would also make it possible to precisely control the lifetime of the entries in the future